libraries/core_libs/storage/src/storage.cpp

```diff
@@ -495,17 +495,12 @@ std::optional<std::pair<uint32_t, uint32_t>> DbStorage::getTransactionPeriod(trx
 }
 
 std::vector<bool> DbStorage::transactionsFinalized(std::vector<trx_hash_t> const& trx_hashes) {
-  std::vector<bool> result;
-  result.reserve(trx_hashes.size());
-
-  DbStorage::MultiGetQuery db_query(shared_from_this(), trx_hashes.size());
-  db_query.append(DbStorage::Columns::trx_period, trx_hashes);
-  auto db_trxs_period = db_query.execute();
-  for (size_t idx = 0; idx < db_trxs_period.size(); idx++) {
-    auto& trx_raw_period = db_trxs_period[idx];
-    result.push_back(!trx_raw_period.empty());
+  std::vector<bool> result(trx_hashes.size(), false);
+  for (size_t i = 0; i < trx_hashes.size(); ++i) {
+    if (exist(toSlice(trx_hashes[i].asBytes()), Columns::trx_period)) {
```

Comment:
So even though multiget is a single call and here we are doing multiple calls of this exist method, you measured it and it is faster to use exist?

Reply:
yop, check the issue I put there result. It is 2x as fast because exist does not fetch data and it is very fast when trx does not exist